iterators_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,59 @@
 #include "impl.h"
 
+#include <sstream>
+#include <iterator>
+#include <algorithm>
+#include <list>
+
+
+std::string reverseWords(const std::string& input)
+{
+    if (input.empty())
+    {
+        return {};
+    }
+
+    std::istringstream inputStream {input};
+    std::ostringstream outputStream;
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es22-dont-declare-a-variable-until-you-have-a-value-to-initialize-it-with